compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -432,8 +432,11 @@ impl<'a> Linker for GccLinker<'a> {
         // insert it here.
         if self.sess.target.is_like_osx {
             self.linker_arg("-dead_strip");
+        // -z ignore
+        // Ignores shared object dependencies that are not referenced as part of the link-edit.
```

Comment:
Are you quoting some linker documentation here? Did that documentation come from Solaris? (That could potentially be a license issue, if so.)

In any case, if this line *is* safe to include, can you please move it inside the else-if block below, so that it's attached to the calls to `linker_arg` rather than being inside the block for osx?

Reply:
good call, it was from Solaris documentation.

I moved the comment into the else-if block and replaced it was a a comment noting the relationship with --as-needed